src/test/java/com/google/sps/CalendarServletTest.java

```diff
@@ -0,0 +1,166 @@
+package com.google.sps;
+
+import static com.google.sps.utils.StringConstants.*;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.sps.servlets.CalendarServlet;
+import com.google.sps.TestUtils;
+import com.google.sps.utils.ServletUtils;
+import com.google.sps.utils.SoyRendererUtils;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.HashSet;
+import javax.management.remote.JMXPrincipal;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+
+@RunWith(JUnit4.class)
+public final class CalendarServletTest extends Mockito {
+  private final static String TEST_POST_EXPECTED_CALENDARID = "mock calendar id";
+  private final static String TEST_GET_USER_EMAIL = "email@example.com";
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig())
+      .setEnvIsAdmin(true).setEnvIsLoggedIn(true);
+  
+  private CalendarServlet servlet;
+  private DatastoreService datastore;
+  private Entity groupEntity;
+  private PrintWriter printWriter;
+  private StringWriter stringWriter;
+
+  @Mock
+  HttpServletRequest request;
+
+  @Mock
+  HttpServletResponse response;
+
+  @Before
+  public void setUp() throws IOException {
+    servlet = new CalendarServlet();
+
+    helper.setUp();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    stringWriter = new StringWriter();
+    printWriter = new PrintWriter(stringWriter);
+
+    groupEntity = TestUtils.createGroupEntity(datastore);
+
+    MockitoAnnotations.initMocks(this);
```

Comment:
Consider moving this to the top of the setUp() function since it's setting up the mocks framework. This'll help it be a bit more future proof in case later, other things in the setUp() need mocks to be ready